src/crypto/CHIPCryptoPALmbedTLS.cpp

```diff
@@ -1217,46 +1217,63 @@ CHIP_ERROR GetNumberOfCertsFromPKCS7(const char * pkcs7, uint32_t * n_certs)
 }
 
 CHIP_ERROR ValidateCertificateChain(const uint8_t * rootCertificate, size_t rootCertificateLen, const uint8_t * caCertificate,
-                                    size_t caCertificateLen, const uint8_t * leafCertificate, size_t leafCertificateLen)
+                                    size_t caCertificateLen, const uint8_t * leafCertificate, size_t leafCertificateLen,
+                                    CertificateChainValidationResult & result)
 {
 #if defined(MBEDTLS_X509_CRT_PARSE_C)
     CHIP_ERROR error = CHIP_NO_ERROR;
-    mbedtls_x509_crt cert_chain;
-    mbedtls_x509_crt root_cert;
-    int result;
+    mbedtls_x509_crt certChain;
+    mbedtls_x509_crt rootCert;
+    int mbedResult;
     uint32_t flags;
 
-    mbedtls_x509_crt_init(&cert_chain);
-    mbedtls_x509_crt_init(&root_cert);
+    VerifyOrReturnError(rootCertificate != nullptr && rootCertificateLen != 0,
```

Comment:
Here as well, start by setting `result` to `kInternalFrameworkError`

Reply:
Fixed in 45c4216